src/webob/acceptparse.py

```diff
@@ -1003,17 +1003,16 @@ def acceptable_offers(self, offers):
             # text/html' (which does not make sense, but is nonetheless valid),
             # and offers is ['text/html']
         ]
-        # sort by offer_index, ascending
-        acceptable_offers_n_quality_factors.sort(key=lambda tuple_: tuple_[2])
-        # (stable) sort by qvalue, descending
         acceptable_offers_n_quality_factors.sort(
-            key=lambda tuple_: tuple_[1], reverse=True
+            key=lambda tuple_: (tuple_[1], -tuple_[2]),
+            reverse=True,
+            # descending sort by (qvalue, -offer_index)
         )
-        # drop offer_index
-        acceptable_offers_n_quality_factors = [
-            (item[0], item[1]) for item in acceptable_offers_n_quality_factors
+        return [
+            (item[0], item[1])
+            for item in acceptable_offers_n_quality_factors
+            # dropping offer_index
```

Comment:
Don't put this inside the list comprehension, it currently causes a bug in black to be exposed which may cause issues when formatting the code base.

Just place it on the line before the `return` please.

Reply:
Do you mean the comment @bertjwregeer?

Reply:
Yes, I mean the comment. See https://github.com/python/black/issues/563 for the issue.

Reply:
Ok, I've made the change.